docs/EVALUATION.md

```diff
@@ -0,0 +1,249 @@
+
+During the core pipeline execution (not including experiment mode pipeline steps), we cannot access future data for evaluation. Also we don't have an evaluation split available. Therefore, we need to use the training data up to the current point in time for evaluation. This is not ideal but it is the only way to evaluate the model in an online fashion.
+
+> The question of how far into the past we want to includes samples remains. More thought is needed before continuing on this.
+
+### Use case 2: offline evaluation
+
+In experiment mode we can potentially utilize the full evaluation split over the whole dataset time range. Being able to "look into the future" is important to determine the performance of a model.
+
+#### [2] Total pipeline performance
+
+We argue that the performance of pipeline w.r.t. a certain evaluation metric is the weighted average of the performance of the most up to date model at each point in time over the whole time range of the pipeline run.
+
+Mind, that we need to weight every model performance at a point in time by the number of samples within that time range.
+
+> E.g. in `model 3` from the Gantt diagram above: assuming our dataset sample distribution is so uneven that 90% of the samples are in the time range between `Trigger3` and `Trigger4`, then the performance of `model 3` should have a higher weight in the pipeline performance than the performance of `model 2`.
+
+It can be argued that the model performance of the newest model in the time span between the ends of two training processes is measured well by the model performance (evaluation result) when evaluating with using all evaluation split samples between the two triggers.
+
+> **Remark/Assumptions**: Instead of considering the time between two model training completions, we will go forward with the time between two triggers (start of training processes). These are not exactly the same, as we would still use the old model while a training is going on (see diagram below). However, we argue that the difference is negligible as (1) we don't expect much drift between the samples directly after the trigger and after the training (close in time). And (2) compared to the time between two triggers, the duration of a training process is rather small and therefore doesn't have a big impact on the evaluation. Note: We do not evaluate on training samples in this model! We can adjust this assumption later on if needed by incorporating the training time into the evaluation strategy.
+
+##### More fine grained view
+
+Accounting for the afore mentioned assumptions that training takes some time and triggers are not at one point in time but specify a training batch (time interval), we can visualize a pipeline run as follows:
+
+
+```mermaid
+gantt
+    title Pipeline run
+    dateFormat YYYY-MM-DD
+    axisFormat %Y
+    section models
+        Trigger1 :milestone, t1, 1930, 0d
+        train_batch_1 :a10, 1928, 2y
```

Comment:
what exactly do you mean by train batch 1? it is different from training. I thought it means training on the first batch of data but I might not fully get the gantt chart yet

Reply:
That represents the interval between the sample with the smallest and biggest timestamps in the batch that is handed to the trainer sever for a new training. I assumed that retraining is only performed on the new samples since the last training, which does not hold for all sampling strategies. I will rename that.